WordPress/gutenberg
# Writing flow: Fix focus loss on toolbar return when last focus element isn't found

## What?


In the Custom HTML block for example, selecting the toolbar Preview or HTML buttons causes the block to re-render its UI elements. When this happens, the writing flow `UseTabNav` hook cannot return focus because that element no longer exists.

## Why?


Fix accessibility bug. Help move forward this PR: #54408.

## How?


If last focus is not possible, I will try to locate the first tabbable in the block similar to what happens in block list `useFocusFirstElement` hook. If no tabbable found within the same block, I will place focus on the block wrapper since it should always be able to receive focus.

## Testing Instructions





1. Insert an image block.
2. Upload or select a media file.
3. Focus the toolbar with Alt+F10.
4. Press Tab to place focus back in the block.
5. Focus should land in the block wrapper since the block no longer is in placeholder state and there are no tabbable elements.

Custom HTML still has some outstanding issues to fix so can't use it in testing currently.

### Testing Instructions for Keyboard


## Screenshots or screencast
